data/network/src/commonMain/kotlin/social/androiddev/common/network/model/Application.kt

```diff
@@ -18,12 +18,14 @@ import kotlinx.serialization.Serializable
 @Serializable
 data class Application(
     // required attributes
+    @SerialName("id") val id: String,
     @SerialName("name") val name: String,
+
+    // client attributes
```

Comment:
Not sure if these client attributes are also optional. Depends in which calls this entity is reused.

Reply:
these client attributes are exactly what we need from this response. We will need to cache/store these and use them in follow-up calls

Reply:
Yes, I know. But the `Application` entity is re-used for other responses without these client attributes.
Like in a status: https://docs.joinmastodon.org/methods/statuses/

So I would change them to be nullable and move it back to the bottom

Reply:
@thebino with regards to _data integrity_ - what are your thoughts on having separate models then in this case? It would make for cleaner/self-documenting models and structure, so I don't need to validate non-null outside of the deserialization